src/harness/constraint_testcase.py

```diff
@@ -0,0 +1,125 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+from datetime import datetime,timedelta
+import logging
+import os
+import time
+import sys
+import sas
+import sas_testcase
+from sas_test_harness import SasTestHarness, GetSasTestHarnessUrl
+
+class ConstraintTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    # Note that we disable the admin.Reset(), to avoid 'unexpected' request
+    # to SAS UUT before a test really starts.
+    # Tests changing the SAS UUT state must explicitly call the SasReset()
+    #self.SasReset()
+    self.sasTHInstances = {}
+
+  def SasReset(self):
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+     # check if any SASTH is started and try to stop it
+     for sasTH in sorted(self.sasTHInstances.itervalues()):
+         sasTH.shutdown()
+
+  def createSASTH(self,sasTHConfig):
+     sasTHSettings = sasTHConfig['sas_hareness']
+     # prepare a HTTPServer Object for each sasTH mentioned in configuration file
+     for item in sasTHSettings:
+        sasTHServer = SasTestHarness(name=item['name'],url=item['url'],
+                                     cert_file=item['cert_file'],key_file=item['key_file'])
+        self.sasTHInstances[item['name']] = sasTHServer
+
+  def configureFADForEachSASTH(self,iterationNumber):
+     '''
+     This method used to configure FAD record for each SAS-TH that includes CBSDactivity based on configuration files
+     and start SAS-TH Http Server instance with configurable host_name and port and ready to serve for Pull Dump request from
+     SAS UUT/Main Test Hareness
+     SAS-TH Http Server will be started automatically as background thread and alive till end of Main Thread/Main Test Hareness
+     SAS-TH Http Server will be stopped automatically during teardown of tests if any
+     :param configDir:
+     :return: None
+     '''
+     def getTimeStamp():
+       '''
+       helper method used to calculate start/end/grant time to populate in FAD records
+       :returns start_time,end_time,grant_expire_time
+       '''
+       end_time = datetime.now()
+       time_delta = timedelta(hours=-24)
+       start_time = end_time.today() + time_delta
+       end_time = str(end_time.replace(microsecond=0).isoformat()) + 'Z'
+       start_time = str(start_time.replace(microsecond=0).isoformat()) + 'Z'
+       time_delta1 = timedelta(hours=24)
+       grant_expire = datetime.now().today() + time_delta1
+       grant_expire = str(grant_expire.replace(microsecond=0).isoformat()) + 'Z'
+       return (start_time,end_time,grant_expire)
+    
+     configDir = 'configs/test_WINNF_FT_S_MCP_1' 
```

Comment:
Please do not statically configure this -- you can automatically read in the config using the @configurable_testcase decorator.